### ISSUE TITLE
Improve the gen_new_probl.sh script

### ISSUE THREAD

**Issue by T-Rex96** [PhilippSchuette/projecteuler#20]
- [x] Add more languages, currently only C++, C and Python are supported (@DanielSchuette for Go and Javascript?)
- [x] It would be pretty cool if the script was able to extract the problem statement from projecteuler.net and substitute it into the generated template file. That probably will require some HTML parsing (@T-Rex96)

**Comment by DanielSchuette** [comment]
I will take a look at it! It would probably be easier to implement the entire thing in Python if we want to add information about the problems/request that information from projecteuler.net. We could think about connecting to one of our user accounts to be able to request the correct solutions, too. But for that, one would need to set password and user name (probably easy in Python, don't know about shell script though).

**Comment by DanielSchuette** [comment]
I started to add `Go` and `JavaScript`. Things I still need to do before we can close this issue (maybe another issue/project for a full-fledged CLI?):
- [ ] add solver to `js`
- [ ] add author, date, and problem number to `go` and `js`

**Comment by DanielSchuette** [comment]
@T-Rex96 can we download input files whenever we encounter an `href` in he problem statement that gets parsed out? This could be another CLI option, e.g. `--dependencies` always tries to download anything that looks like an input/data file.

**Comment by T-Rex96** [comment]
Cool idea, I'll try and add this later tonight

**Comment by T-Rex96** [comment]
Done, but I've only tested this with problem 22, do we know of any other problem statements with input files that we could use for testing this?

**Comment by DanielSchuette** [comment]
Unfortunately not. We can keep this issue open until we have another working example. Problem 22 works well though, great job!

**Comment by DanielSchuette** [comment]
Should we relocate all templates to a central `template` dir? @T-Rex96 @PhilippSchuette

**Comment by T-Rex96** [comment]
Would probably make sense 🤷🏻‍♂️
